app/assets/javascripts/controllers/SearchCtrl.coffee

```diff
@@ -0,0 +1,29 @@
+@bruseApp.controller 'SearchCtrl', ['FileHandler', '$scope', '$http', (FileHandler, $scope, $http) ->
+  $scope.files = []
+
+  # watches the search input field for changes
+  $scope.$watch "search", () ->
+    if($scope.search != "")
+      # send a search request to the server
+      $http.get('/search/'+$scope.search)
+        .then((response) ->
+          if response.data.files.length > 0
```

Comment:
Is this check necessary? If the repeat directive receives an empty array, it simply ignores doesn't render it.